Sources/KlaviyoSwift/AppContextInfo.swift

```diff
@@ -30,17 +48,76 @@ struct AppContextInfo {
         "\(osName) \(osVersion)"
     }
 
-    init(excutable: String = DEFAULT_EXECUTABLE,
+    init(executable: String = DEFAULT_EXECUTABLE,
          bundleId: String = DEFAULT_BUNDLE_ID,
          appVersion: String = DEFAULT_APP_VERSION,
          appBuild: String = DEFAULT_APP_BUILD,
+         appName: String = DEFAULT_APP_NAME,
          version: OperatingSystemVersion = DEFAULT_OS_VERSION,
-         osName: String = DEFAULT_OS_NAME) {
-        self.excutable = excutable
+         osName: String = DEFAULT_OS_NAME,
+         manufacturer: String = DEFAULT_MANUFACTURER,
+         deviceModel: String = DEFAULT_DEVICE_MODEL,
+         deviceId: String = UIDevice.current.identifierForVendor?.uuidString ?? "") {
+        self.executable = executable
         self.bundleId = bundleId
         self.appVersion = appVersion
         self.appBuild = appBuild
+        self.appName = appName
         self.version = version
         self.osName = osName
+        self.manufacturer = manufacturer
+        self.deviceModel = deviceModel
+        self.deviceId = deviceId
+
+        switch UIDevice.current.pushEnvironment {
+        case .development:
+            environment = "debug"
+        case .production:
+            environment = "release"
+        case .unknown:
+            #if DEBUG
+            environment = "debug"
+            #else
+            environment = "release"
+            #endif
+        }
+    }
+}
+
+extension UIDevice {
+    public enum PushEnvironment: String {
+        case unknown
```

Comment:
I know nothing on how this works, but I'm wondering why we would want an `unknown` environment. If APNS gives us nothing, can we assume thats a device running on prod and default to that?

Reply:
i used this as a fallback to then use the DEBUG flag

Reply:
I think this is indicating we can't determine the type.

Reply:
Aaah I see above. Thanks - makes perfect sense to me

Reply:
Err the environment.

Reply:
The question is what would we do with these tokens?

Reply:
I suspect we'd only hit this if the app that doesn't use push notifications at all. In which case I'll just fall back on using the DEBUG compiler flag. 
As far as what we send the server I'll still only ever send `debug` or `release`

Reply:
ok cool